fetch/h1-parsing/lone-cr.window.js

```diff
@@ -0,0 +1,20 @@
+// These tests expect that a network error is returned if there's a CR that is not immediately
+// followed by LF before reaching message-body.
+//
+// No browser does this currently, but Firefox does treat it equivalently to a space which gives
+// hope.
+
+[
+  "HTTP/1.1\r200 OK\n\nBODY",
+  "HTTP/1.1 200\rOK\n\nBODY",
+  "HTTP/1.1 200 OK\n\rHeader: Value\n\nBODY",
+  "HTTP/1.1 200 OK\nHeader\r: Value\n\nBODY",
+  "HTTP/1.1 200 OK\nHeader:\r Value\n\nBODY",
+  "HTTP/1.1 200 OK\nHeader: Value\r\r\nBODY",
```

Comment:
Suggest the following:

* \r before the body, since the final linebreak is potentially special.  e.g.
  "HTTP/1.1 200 OK\nHeader: Value\n\rBODY"
* \r without a body, which is also potentially a bit special.
   "HTTP/1.1 200 OK\nHeader: Value\n\r"
* One with just \r after a value (first line is potentially special, so best to have a \r at a reasonable place after a header line)
  "HTTP/1.1 200 OK\nHeader: Value\r\nBody"